### PR TITLE
Clean up side nav + code excerpts for cookbook/persistence and cookbook/maintenance

### PR DESCRIPTION
**Pull request by khanhnwin**
_Description of what this PR is changing or adding, and why:_ The /development page was wrong and it shouldn't have even existed according to the structure for all of the other titles. Removed the page + its permalink. Also removed permalinks for data-and-backend/state-mgmt/ and data-and-backend/state-mgmt/

_Issues fixed by this PR (if any):_ #6758 

CC @domesticmouse @sfshaza2 I haven't had a chance to look into why each of the side nav titles still have an anchor link, but it seems to all match up for now. 

## Presubmit checklist
- [X] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [X] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [X] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/_data/sidenav.yml [hunk 1]
```diff
@@ -117,10 +116,8 @@
         - title: Widget catalog
           permalink: /development/ui/widgets
     - title: Data & backend
-      permalink: /development/data-and-backend
```
Comment (domesticmouse) [inline src/_data/sidenav.yml:120]: Does this have an empty .md file backing it

File: src/_data/sidenav.yml [hunk 2]
```diff
@@ -117,10 +116,8 @@
...
       children:
         - title: State management
-          permalink: /development/data-and-backend/state-mgmt
```
Comment (domesticmouse) [inline src/_data/sidenav.yml:123]: Or this one

**Review by domesticmouse — APPROVED** [review]

**Review by sfshaza2 — APPROVED** [review]
Thanks, @khanhnwin! lgtm

**Comment by khanhnwin** [comment]
Oops I added some commits that I didn't expect to. I'll fix that!

File: src/_data/sidenav.yml (hunk 1 above)
Comment (khanhnwin) [reply src/_data/sidenav.yml:120]: Yes it does! https://docs.flutter.dev/development/data-and-backend

File: src/_data/sidenav.yml (hunk 2 above)
Comment (khanhnwin) [reply src/_data/sidenav.yml:123]: Yep! https://docs.flutter.dev/development/data-and-backend/state-mgmt

**Comment by khanhnwin** [comment]
Thanks Shams! Merged :)

**Comment by sfshaza2** [comment]
LOL! I see what you mean!

**Pull request merged**